src/components/SelectBox.js

```diff
@@ -83,6 +94,10 @@ SelectBox.propTypes = {
   /* eslint-enable react/no-unused-prop-types */
   onChange: PropTypes.func.isRequired, // (selectedId: string) => any
   id: PropTypes.string,
+  validationState: PropTypes.oneOfType([
```

Comment:
Looks like `validationState` should just be a list of possible values, and have it be optional (default 'default').  So like:

```jsx
validationState: PropTypes.oneOf([ 'default', 'error' ]),
```

Then you can add a default value like this (add below the propTypes):

```jsx
SelectBox.defaultProps = {
  validationState: 'default',
}
```

Reply:
@sjd78 it seems like it doesn't work in that case:
![image](https://user-images.githubusercontent.com/64131213/94520749-7b4b8080-0235-11eb-9763-8fd1cd9d2356.png)
In case of using `OR`/`AND` statements `false` is possible returned value, so we need to consider it in our `PropTypes`.
WDYT about using one of the options below:
```
validationState: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.oneOf([ false ]),
  ]),
```
```
  validationState: PropTypes.oneOf([ false, 'default', 'error' ]),
```
Do you have another suggestion?

Reply:
I understand you refer to following usage pattern:
`validationState={!sd.isOk && 'error'}`
It's an universal problem of dynamic optional attribute. There are several ways to solve that (i.e. [this thread])(https://stackoverflow.com/questions/31163693/how-do-i-conditionally-add-attributes-to-react-components).
Actually I would rather avoid embedding that in type definition (sorry for not spotting that during first review). 
We could i.e. just fall-back to default: `validationState={!sd.isOk && 'error' || 'default'}`

Reply:
I prefer the trinary statement too, in my first PRs @sjd78 ask me every time to change trinary statements to that pattern and in the project most of the time we are using that pattern:
> validationState={!sd.isOk && 'error'}

So IMHO we need to support using this pattern or change it everywhere in the project to keep consistency.
WDYT?

Reply:
I think both options are okay if t's just about PropTypes. It doesn't have any impact on functionality, right?

Reply:
Ok, so....

While it is true that PropTypes don't break things at run time, we should not be seeing any errors when running in dev mode.  In prod mode, PropTypes warnings are not added.  Runtime type errors tend to indicate that something will go very wrong, very quietly.

Yes I usually like to see a `condition && something` rather than `condition ? something : undefined` but it is more context sensitive.  When doing components that are optionally rendered, the `condition && <Component>` is my favorite.  When doing attributes, could go either way, but really depends on what the attribute values need to be.  With an attribute that needs to be `undefined` (PropTypes not required) or one of a set of possible values (PropTypes.oneOf), the ternary makes sense.  So here, it would look like `validationState={sd.isOk ? 'default' : 'error'}` if the PropTypes doesn't allow a false value that maps to 'default'.

If `SelectBox` will only have 2 states - normal and error - then the attribute can just be a boolean `valid`.  If we want to have >2 states, then a enum of strings makes sense.  That just means the optional attribute must be present and a string or must be undefined and then get a default value.

Finally, using: `PropTypes.oneOf([ false, 'default', 'error' ])` and handling `false` the same as `undefined` / use the default value, should be ok.